core/src/test/java/org/apache/calcite/util/format/FormatElementEnumTest.java

```diff
@@ -28,150 +28,94 @@
 
   @Test void testMM() {
-    final String date = "2014-09-30T10:00:00Z";
-    StringBuilder ts = new StringBuilder();
-    FormatElementEnum.MM.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("09"));
+    assertFormatElement(FormatElementEnum.MM, "2014-09-30T10:00:00Z", "09");
   }
 
   @Test void testMON() {
-    final String date = "2014-09-30T10:00:00Z";
-    StringBuilder ts = new StringBuilder();
-    FormatElementEnum.MON.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("Sep"));
+    assertFormatElement(FormatElementEnum.MON, "2014-09-30T10:00:00Z", "Sep");
   }
 
   @Test void testQ() {
-    final String date = "2014-09-30T10:00:00Z";
-    StringBuilder ts = new StringBuilder();
-    FormatElementEnum.Q.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("3"));
+    assertFormatElement(FormatElementEnum.Q, "2014-09-30T10:00:00Z", "3");
   }
 
   @Test void testMS() {
-    final String date = "2014-09-30T10:00:00Z";
-    StringBuilder ts = new StringBuilder();
-    FormatElementEnum.MS.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("000"));
+    assertFormatElement(FormatElementEnum.MS, "2014-09-30T10:00:00Z", "000");
   }
 
   @Test void testSS() {
-    final String date = "2014-09-30T10:00:00Z";
-    StringBuilder ts = new StringBuilder();
-    FormatElementEnum.SS.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("00"));
+    assertFormatElement(FormatElementEnum.SS, "2014-09-30T10:00:00Z", "00");
   }
 
   @Test void testWM() {
-    final String date = "2014-09-30T10:00:00Z";
-    StringBuilder ts = new StringBuilder();
-    FormatElementEnum.W.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("5"));
+    assertFormatElement(FormatElementEnum.W, "2014-09-30T10:00:00Z", "5");
   }
 
   @Test void testWW() {
-    final String date = "2014-09-30T10:00:00Z";
-    StringBuilder ts = new StringBuilder();
-    FormatElementEnum.WW.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("40"));
+    assertFormatElement(FormatElementEnum.WW, "2014-09-30T10:00:00Z", "40");
   }
 
   @Test void testYY() {
-    final String date = "2014-09-30T10:00:00Z";
-    StringBuilder ts = new StringBuilder();
-    FormatElementEnum.YY.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("14"));
+    assertFormatElement(FormatElementEnum.YY, "2014-09-30T10:00:00Z", "14");
   }
 
   @Test void testYYYY() {
-    final String date = "2014-09-30T10:00:00Z";
+    assertFormatElement(FormatElementEnum.YYYY, "2014-09-30T10:00:00Z", "2014");
+  }
+
+  private void assertFormatElement(FormatElementEnum formatElement, String date, String expected) {
     StringBuilder ts = new StringBuilder();
-    FormatElementEnum.YYYY.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("2014"));
+    formatElement.format(ts, Date.from(Instant.parse(date)));
```

Comment:
I noticed that other test methods are placed above. Do we need to comply with this specification?

Reply:
Well, it seems irrelevant.